dts: arm: Add CAN1 and CAN2 nodes to stm32f207

camponotus opened the pull request:
Add CAN1 and CAN2 device tree nodes for STM32F207

Signed-off-by: Taeyoun Park <nvnv0422@gmail.com>

djiatsaf-st reviewed (commented):
Non blocking comments
LGTM

camponotus commented on dts/arm/st/f2/stm32f207.dtsi:
```diff
@@ -34,5 +34,23 @@
+			reg = <0x40006400 0x400>;
+			interrupts = <19 0>, <20 0>, <21 0>, <22 0>;
+			interrupt-names = "TX", "RX0", "RX1", "SCE";
+			clocks = <&rcc STM32_CLOCK(APB1, 25U)>;
```
Based on what I've checked, adding a suffix to the corresponding item seems to be the common convention, at least for the st series. Would you still prefer to have it changed?

djiatsaf-st replied:
Yes please,  i prefer to have it without suffix .

camponotus replied:
Alright! I've removed the suffix.
I can update other DTS files too if needed, but I’ll leave them as-is for now.

camponotus commented:
Alright! I've removed the suffix.
I can update other DTS files too if needed, but I’ll leave them as-is for now.

gautierg-st reviewed (changes requested):
Could you also enable a CAN instance in the Nucleo F207ZG board dts for test purposes (if it is possible to do it without pin conflict)?

camponotus commented:
> Could you also enable a CAN instance in the Nucleo F207ZG board dts for test purposes (if it is possible to do it without pin conflict)?

I was able to test the CAN bus on the Nucleo F207ZG board and confirmed it was working fine(no nodes removed or pins changed). I configured the timing to 500kbps, 80.0%, and it worked as expected. Is my understanding correct?
&can1 {
    status = "okay";
    pinctrl-0 = <&can1_rx_pd0 &can1_tx_pd1>;
    pinctrl-names = "default";
    label = "CAN1";
};

camponotus commented:
I think it’s better to have it as a single commit, so I force-pushed it. Please check it.

etienne-lms reviewed (commented):
Could you add in the commit message that you also enable CAN1 and CAN2 on nucleo_f207zg board?

Maybe it would be better to have 2 commits: the first one for the SoC DTSI and a second one dedicated to the nucleo_f207zg board DTS change. This last commit could add `can` to the `supported` features in **nucleo_f207zg.yml** file. Also could add the CAN1/CAN2 pins musing info in **nucleo_f207zg/doc/index.rst** documentation file.

camponotus committed:
dts: arm: Add CAN1 and CAN2 nodes to stm32f207

Add CAN1 and CAN2 device tree nodes for STM32F207

Signed-off-by: Taeyoun Park <nvnv0422@gmail.com>

camponotus committed:
boards: st: Enable CAN1 and CAN2 on nucleo_f207zg

Enable CAN1 and CAN2 in nucleo_f207zg.dts. Update supported
features in YAML and the board documentation accordingly.

Signed-off-by: Taeyoun Park <nvnv0422@gmail.com>

etienne-lms reviewed (commented):
Indeed, first commit message needs to be fixed:
```patch
-Signed-off-by: Taeyoun Park nvnv0422@gmail.com
+Signed-off-by: Taeyoun Park <nvnv0422@gmail.com>
